### DISCUSSION TITLE
Document upgrade and downgrade processes

### DISCUSSION THREAD

**Discussion by zamazan4ik** [rook/rook#10682] [General]
Hi.

After reading the documentation it is not clear for me, how upgrade and downgrade processes should be performed over the solution. Could you please describe somewhere in the documentation:

- How to upgrade Rook to the newer version (hopefully with no downtime)
- How to downgrade Rook to the older version (hopefully with no downtime)
- Maybe some compatibility policies and notes between versions

Thanks in advance!

**Comment by travisn** [comment]
@zamazan4ik Did you see these topics?
- Health verification for upgrade: https://rook.io/docs/rook/v1.9/Upgrade/health-verification/
- Rook upgrades: https://rook.io/docs/rook/v1.9/Upgrade/rook-upgrade/
- Ceph upgrades: https://rook.io/docs/rook/v1.9/Upgrade/ceph-upgrade/

**Reply by zamazan4ik** [reply]
@travisn Yep, I have read them. I was not sure that Rook upgrade could be performed without downtime  (according to the link - https://rook.io/docs/rook/v1.9/Upgrade/rook-upgrade/) but as far as I understand, thi point is covered in https://rook.io/docs/rook/v1.9/Upgrade/health-verification/).

However, I did not find any mention about downgrade support? Is it supported? How the downgrade process should be performed?

**Reply by travisn** [reply]
Downgrade is not tested so it is not officially supported. But in general, the steps would be the same for downgrade as for upgrade, you just apply the desired versions and then rook will start them. I would expect downgrading between patch releases is generally ok, but downgrading across major/minor versions is more risky.
